test/plugin_functional/test_suites/application_links/redirect_app_links.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import url from 'url';
+import expect from '@kbn/expect';
+import { PluginFunctionalProviderContext } from '../../services';
+import '../../plugins/core_app_status/public/types';
+
+declare global {
+  interface Window {
+    __nonReloadedFlag?: boolean;
+  }
+}
+
+const getKibanaUrl = (pathname?: string, search?: string) =>
+  decodeURIComponent(
+    url.format({
+      protocol: 'http:',
+      hostname: process.env.TEST_KIBANA_HOST || 'localhost',
+      port: process.env.TEST_KIBANA_PORT || '5620',
+      pathname,
+      search,
+    })
+  );
+
+// eslint-disable-next-line import/no-default-export
+export default function ({ getService, getPageObjects }: PluginFunctionalProviderContext) {
+  const PageObjects = getPageObjects(['common']);
+  const browser = getService('browser');
+  const testSubjects = getService('testSubjects');
+
+  const setNonReloadedFlag = async () => {
+    return browser.executeAsync(async (cb: Function) => {
+      window.__nonReloadedFlag = true;
+      cb();
+    });
+  };
+  const wasReloaded = async (): Promise<boolean> => {
+    return browser.executeAsync<boolean>(async (cb) => {
+      const reloaded = window.__nonReloadedFlag !== true;
+      cb(reloaded);
+      return reloaded;
```

Comment:
not needed?

Reply:
Unfortunately It is, because our `executeAsync` signature is broken (and so is webdriver's but in a different way), but I got tired of adding the comment on every `browser.executeAsync` tbh.

https://github.com/elastic/kibana/blob/13138c05c4a09b6db79117d6e55628cfee1e09fd/test/functional/services/common/browser.ts#L482-L485

Reply:
Agree. But TS doesn't work well with WebDriver implementation when a callback is always the last argument passed in a function.
We could change executeAsync signature to accept 1 additional argument  at max:
```typescript
async executeAsync<T = any>(
  fn: <T>(cb: (value: T) => void) => void | Promise<void>,
): Promise<T>;
async executeAsync<T = any, Context extends Record<string, any> = Record<string, any>>(
  fn: <T>(context: Context, cb: (value: T) => void) => void | Promise<void>,
  context: Context,
): Promise<T>;
```
There are 13 tests with `executeAsync` in the code base, it doesn't look like a significant effort. @elastic/kibana-qa Any objections we do this change in a separate PR?

Reply:
I'll defer to @dmlemeshko on this.

Reply:
> @elastic/kibana-qa Any objections we do this change in a separate PR?

No problem from my side.